src/main/java/com/solbeg/nuserservice/controller/AuthController.java

```diff
@@ -16,6 +23,19 @@ public class AuthController {
 
     @PostMapping("/login")
     public TokenResponse login(@RequestBody AuthParamsModel params) {
-        return authService.login(params);
+        Optional<User> optionalUser = authService.findUserByEmail(params);
+        if (optionalUser.isPresent()) {
+            if (authService.correctPassword(params, optionalUser.get())) {
```

Comment:
avoid inner "if" as much as possible
here you can move user fetching in private method
User  user = findUserByEmailOrThrowException(params.getEmail())